include/exadg/utilities/exceptions.h

```diff
@@ -0,0 +1,45 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2022 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_UTILITIES_EXCEPTIONS_H_
+#define INCLUDE_EXADG_UTILITIES_EXCEPTIONS_H_
+
+#include <deal.II/base/exceptions.h>
+
+namespace ExaDG
+{
+DeclExceptionMsg(ExcNotImplemented,
+                 "You are trying to use functionality in ExaDG that is "
+                 "currently not implemented. In many cases, this indicates "
+                 "that there simply didn't appear much of a need for it, or "
+                 "that the author of the original code did not have the "
+                 "time to implement a particular case. If you hit this "
+                 "exception, it is therefore worth the time to look into "
+                 "the code to find out whether you may be able to "
+                 "implement the missing functionality. If you do, please "
+                 "consider providing a patch to the ExaDG development "
+                 "sources (see the ExaDG Wiki page on how to contribute).");
```

Comment:
Just a comment, but this smells very slightly like plagiarism when we use a verbatim copy of the text in deal.II (with only the package names exchanged) without explicitly saying that this is a copy of the respective `ExcNotImplemented` in deal.II.

Reply:
Is it with the following comment better?

https://github.com/exadg/exadg/blob/e6c4599a1a41e0eb10891fc0cd18b899a12546c1/include/exadg/utilities/exceptions.h#L30

I really don't care what the message is. The PR only shows how to introduce new exceptions.

Reply:
Yes, no need to adapt it word-by-word, given that the message is also very wordy ...

Let's collect ideas regarding which information such a message should provide in order to help un-experienced users ...

Reply:
Some ideas from my side: The deal.ii-message is somewhat written from the perspective of developers. In my opinion, the message should mention the possibility that the user has chosen a combination of parameters that is actually not intended. Note that it might be quite difficult to explicitly assert all invalid parameter combinations (even though we try this in `Parameters::check()`). Hence, the exception message could provide the suggestion to carefully check the input parameters again. So in some cases such a message is not a hint to implement additional functionality.  I believe `ExcNotImplemented` is still the correct type of exception, since such problems can not be deduced from the implementation in a single file, but need to take into account the interaction of modules.

Reply:
I agree, we should have the first sentence indicating that some combination of features either has not been implemented, or does not make sense in combination.